docs/development/DuckTyping.md

```diff
@@ -491,6 +498,284 @@ AMD Ryzen 9 5950X, 1 CPU, 32 logical and 16 physical cores
+object originalObject;
+var proxy = originalObject.DuckCast<IProxyMyHandler>();
+var chainedProxy = proxy.Configuration
+```
+
+In this situation, you have access to the original `object`, so you can check whether it's `null` directly. Also in the above example, if `originalObject` is null, `DuckCast<T>`() will also return `null`.
+
+By implication this _also_ means that `proxy.Instance` will _never_ be `null` in this case, because we only create a proxy if the instance is non-null.
+
+> Yup, that's right, this is _the exact opposite_ to the rules for generic constraint proxies 😅 
+
+The `chainedProxy` behaves similarly. If the underlying property `originalObject` is `null`, then `proxy.Configuration` will _also_ be `null`. And similarly, if `chainedProxy` is _not_ `null`, then `chainedProxy.Instance` will _also_ not be `null` by definition.
+
+
+#### 3. Using `[DuckCopy]` `struct` proxies in `DuckCast<T>` or in duck chained properties
+
+The final scenario is similar to the above, but it uses `[DuckCopy]` `struct` instead of interface. 
+
+> A `[DuckCopy]` `struct`, as the name implies, creates a `struct` and _copies_ the values from the source object into the struct. Their features are limited because of this (e.g. you can't call methods or access the original object) but they are generally faster and lower allocation.  
+
+As an example, lets create the interface proxies as `[DuckCopy]` `struct` instead:
+
+```csharp
+[DuckCopy] // Add the attributes
+public struct ProxyMyHandlerStruct
+{
+    public string Name; // Note these must be FIELDS not properties
+    public ProxyMyHandlerConfiguration Configuration; // duck chaining struct
+}
+
+[DuckCopy]
+public struct ProxyMyHandlerConfiguration 
+{
+    public int MaxConnections;
+}
+```
+
+and lets use it in the same way
+
+And using it like this:
+
+```csharp
+object originalObject;
+var proxy = originalObject.DuckCast<ProxyMyHandlerStruct>();
+var chainedProxy = proxy.Configuration
+```
+
+As before, if `originalObject` is `null` you can test that directly, and similarly `proxy` will be `null` (because `DuckCast<T>` returns `T?`, i.e. a nullable). Where things are different is with a duck-chained `[DuckCopy]` struct.
+
+In the above example, the `ProxyMyHandlerStruct.Configuration` field is of type `ProxyMyHandlerConfiguration`, which is a `struct`. Therefore _it can never be `null`_. If the underlying `originalObject.Configuration` property was null, `proxy.Configuration` will be the `default` value, so `proxy.Configuration.MaxConnections == 0`! 
+
+What's more, the `[DuckCopy]` `struct` does _not_ implement `IDuckType`. So with the above duck-types, there's no way to detect when `originalObject.Configuration` is `null`! This is a problem, as it could mean accidentally using the `default` value for `MaxConnections` (i.e. `0`) when in practice the `Configuration` property was `null`. 
+
+The solution to this problem, is to always define duck-chained `[DuckCopy` `struct`s as nullable:
```

Comment:
```suggestion
The solution to this problem, is to always define duck-chained `[DuckCopy]` `struct`s as nullable:
```

Reply:
also remove the comma in "The solution to this problem is to..."